topicgenerator/src/main/java/io/dockstore/topicgenerator/helper/AIModel.java

```diff
@@ -0,0 +1,60 @@
+package io.dockstore.topicgenerator.helper;
+
+import java.util.Optional;
+
+/**
+ * An AI model that generates topics.
+ */
+public abstract class AIModel {
```

Comment:
In hierarchies like this, it can be useful to define the core interface as an `interface`, ala:
```
public interface AIModel {
    String getModelName();
    double getPricePer1kInputTokens();
    double getPricePer1kOutputTokens();
    double getMaxContentLength();
    [...]
}
```
Then, you can define the "utility" base abstract class as follows:
```
public abstract class BaseAIModel implements AIModel {
     // implementations of basic functionality
} 
```
Model-specific classes can then extend `BaseAIModel:
```
public class AnthropicClaudeModel extends BaseAIModel {
    // claude-specific implementation
}
```
What does this buy you?  Amongst other things, if you want to implement a new `AIModel` that works differently than the derivatives of `BaseAIModel` - say it wraps another `AIModel`, or combines several `AIModels`, or whatever - you can simply implement `AIModel` directly, and not have to deal with all the stuff in `BaseAIModel` that isn't needed.